hnn_core/tests/test_network.py

```diff
@@ -83,6 +83,49 @@ def test_network_models():
         assert np.all(np.diff(k_gbar, n=2) > 0)  # positive 2nd derivative
 
 
+def test_network_cell_positions():
+    """"Test manipulation of cell positions in the network object"""
+
+    net = jones_2009_model()
+    assert np.isclose(net._inplane_distance, 1.)
+    assert np.isclose(net._layer_separation, 1307.4)
+
+    net.set_cell_positions(inplane_distance=2.)
+    # check that in-plane distance is now 2. for the default 10 x 10 grid
+    assert np.allclose(  # x-coordinate jumps every 10th gid
+        np.diff(np.array(net.pos_dict['L5_pyramidal'])[9::10, 0], axis=0), 2.)
+    assert np.allclose(  # test first 10 y-coordinates
+        np.diff(np.array(net.pos_dict['L5_pyramidal'])[:9, 1], axis=0), 2.)
+
+    net.set_cell_positions(layer_separation=1000.)
+    # check that layer separation has changed (L5 is zero)
+    assert np.isclose(net.pos_dict['L2_pyramidal'][0][2], 1000.)
+
+    with pytest.raises(ValueError,
+                       match='Number of pyramidal cells in each direction'):
+        net.set_cell_positions(n_pyr_y=0)
+    with pytest.raises(ValueError,
+                       match='Number of pyramidal cells in each direction'):
+        net.set_cell_positions(n_pyr_y=0)
+    with pytest.raises(ValueError,
+                       match='In-plane distance must be positive'):
+        net.set_cell_positions(inplane_distance=0.)
+    with pytest.raises(ValueError,
+                       match='Layer separation must be positive'):
+        net.set_cell_positions(layer_separation=0.)
+
+    # check that the origin of the drive cells matches the new 'origin'
```

Comment:
why would the origin change when `set_cell_positions` is called?

Reply:
```python
    # As the network dimensions increase, so does the center-of-mass of the
    # grid points, which is where all hnn drives should be located. The lamtha-
    # dependent weights and delays of the drives are calculated with respect to
    # this origin.
```

Reply:
humm ... I guess this means the neurons are not symmetrically distributed around origin. Side effect of having the origin defined based on the positions of the neurons rather than having the positions based on the origin. I find this a bit weird but can live with it for now ... let's have users complain about it